scripts/scil_tractogram_point_math.py

```diff
@@ -0,0 +1,182 @@
+                   help='Name of the resulting data_per_point or '
+                   'data_per_streamline to be saved in the output '
+                   'tractogram. (Default: %(default)s)')
+    p.add_argument('out_tractogram', metavar='OUTPUT_FILE',
+                   help='The file where the remaining streamlines '
+                        'are saved.')
+
+    p.add_argument('--endpoints_only', action='store_true', default=False,
+                   help='If set, will only perform operation on endpoints \n'
+                   'If not set, will perform operation on all streamline \n'
+                   'points.')
+
+    add_reference_arg(p)
+    add_verbose_arg(p)
+    add_overwrite_arg(p)
+    add_bbox_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    if args.verbose:
+        logging.getLogger().setLevel(logging.INFO)
+
+    assert_inputs_exist(parser, args.in_tractogram)
+    assert_outputs_exist(parser, args, args.out_tractogram)
+
+    # Load the input files.
+    logging.info("Loading file {}".format(args.in_tractogram))
+    sft = load_tractogram_with_reference(parser, args, args.in_tractogram)
+
+    if len(sft.streamlines) == 0:
+        logging.info("Input tractogram contains no streamlines. Exiting.")
+        return
+
+    if len(args.in_dpp_name[0]) != len(args.out_name[0]):
+        parser.error('The number of in_dpp_names and out_names must be '
+                     'the same.')
+
+    data_per_point = {}
+    data_per_streamline = {}
+    for in_dpp_name, out_name in zip(args.in_dpp_name[0],
+                                     args.out_name[0]):
+        # Check to see if the data per point exists.
+        if in_dpp_name not in sft.data_per_point:
+            logging.info('Data per point {} not found in input tractogram.'
+                         .format(in_dpp_name))
+            return
+
+        # check size of first streamline data_per_point
+        if sft.data_per_point[in_dpp_name][0].shape[1] == 1:
+            is_singular = True
+        else:
+            is_singular = False
+
+        if args.operation == 'correlation' and is_singular:
+            logging.info('Correlation operation requires multivalued data per '
+                         'point. Exiting.')
+            return
+
+        # Perform the requested operation.
+        if not is_singular:
+            if args.operation == 'correlation':
+                logging.info('Performing {} across endpoint data.'.format(
+                    args.operation))
+                new_dps = perform_streamline_operation_on_endpoints(
+                    args.operation, sft, in_dpp_name)
+
+                # Adding data per streamline to new_sft
+                data_per_streamline[out_name] = new_dps
+            else:
+                # Results in new data per point
+                logging.info(
+                    'Performing {} on data from each streamine point.'.format(
```

Comment:
I would clarify: `'Performing {} on data from each streamine point and saving as new dpp {}'`